core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -987,34 +989,29 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
   }
 
   //validates the expressions in the method configuration, taking into account optional inputs
-  private def validateMethodConfiguration(methodConfiguration: MethodConfiguration, dataAccess: DataAccess): ReadWriteAction[ValidatedMethodConfiguration] = {
-    withMethodInputs(methodConfiguration, userInfo) { gatherInputsResult =>
-      val vmc = ExpressionValidator.validateAndParseMCExpressions(methodConfiguration, gatherInputsResult, allowRootEntity = methodConfiguration.rootEntityType.isDefined, dataAccess)
-      DBIO.successful(vmc)
-    }
-  }
-
-  def createMCAndValidateExpressions(workspaceContext: SlickWorkspaceContext, methodConfiguration: MethodConfiguration, dataAccess: DataAccess): ReadWriteAction[ValidatedMethodConfiguration] = {
-    dataAccess.methodConfigurationQuery.create(workspaceContext, methodConfiguration) flatMap { _ =>
-      validateMethodConfiguration(methodConfiguration, dataAccess)
-    }
+  private def validateMethodConfiguration(methodConfiguration: MethodConfiguration, workspaceContext: SlickWorkspaceContext): Future[ValidatedMethodConfiguration] = {
+    val entityProvider = getEntityProviderForMethodConfig(workspaceContext, methodConfiguration)
+    for {
+      gatherInputsResult <- gatherMethodConfigInputs(methodConfiguration)
+      vmc <- entityProvider.expressionValidator.validateAndParseMCExpressions(methodConfiguration, gatherInputsResult, allowRootEntity = methodConfiguration.rootEntityType.isDefined)
+    } yield vmc
   }
 
-  def updateMCAndValidateExpressions(workspaceContext: SlickWorkspaceContext, methodConfigurationNamespace: String, methodConfigurationName: String, methodConfiguration: MethodConfiguration, dataAccess: DataAccess): ReadWriteAction[ValidatedMethodConfiguration] = {
-    dataAccess.methodConfigurationQuery.update(workspaceContext, methodConfigurationNamespace, methodConfigurationName, methodConfiguration) flatMap { _ =>
-      validateMethodConfiguration(methodConfiguration, dataAccess)
-    }
+  private def getEntityProviderForMethodConfig(workspaceContext: SlickWorkspaceContext, methodConfiguration: MethodConfiguration) = {
+    // TODO: user method config root entity to figure this out
```

Comment:
Haven't quite gotten to everything yet, but wasn't sure what this was referring to.